tests/test_user_permission.py

```diff
@@ -0,0 +1,146 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2019 Open Education Polito.
+#
+# fare is free software; you can redistribute it and/or modify it under the
+# terms of the MIT License; see LICENSE file for more details.
+
+import json
+
+import requests
+from flask_security import current_user, login_user
+
+
+def test_anonymous_user_get_page(app):
+    """
+    Test purpose:
+    an anonymous user perform a GET to the view for granting permissions
+
+    Input:
+    non authenticated user
+
+    Output:
+    GET of the permission page to change permission with status code 301
+    """
+
+    url = "https://localhost:5000/grant_staff/grant"
+    r = requests.get(url, verify=False, allow_redirects=False)
+    assert r.status_code == 302
+
+
+def test_staff_get_page(app, users):
+    """
+    Test purpose:
+    a user with staff permission can perform a GET
+    to change permission page
+
+    Input:
+    one user who perform a GET with staff permission
+    to the change staff permission page
+
+    Output:
+    GET return grant permission page with status code 200
+    """
+
+    url = "https://localhost:5000/grant_staff/grant"
+    staff = users['staff']
+    login_user(staff)
+    r = requests.get(url, verify=False)
+    assert r.status_code == 200
```

Comment:
I don't think this is exactly what we want to test. 
This is the HTTP code returned which is OK, however I believe this is just *part* of the test: the most important one should be to check whether the staff role has been granted or not. 
I'd like to see something like:
```
assert user !staff
perform actions
assert user staff or still !staff (depending on the actions performed)
```
What do you think? We have direct DB access so we can query it and see what the actions really do